task/thermal/src/bsp/sidecar_bcd.rs

```diff
@@ -114,13 +113,27 @@ impl Bsp {
             3 => 1,
             _ => panic!(),
         };
-        FanControl::Max31790(controller, fan_physical.try_into().unwrap_lite())
+        Ok(FanControl::Max31790(
+            controller.try_initialize()?,
+            fan_physical.try_into().unwrap_lite(),
+        ))
     }
 
-    pub fn for_each_fctrl(&self, mut fctrl: impl FnMut(FanControl<'_>)) {
+    pub fn for_each_fctrl(
+        &mut self,
+        mut fctrl: impl FnMut(FanControl<'_>),
+    ) -> Result<(), ControllerInitError> {
+        let mut last_err = Ok(());
         // Run the function on each fan control chip
-        fctrl(self.fan_control(0.into()));
-        fctrl(self.fan_control(4.into()));
+        match self.fan_control(0.into()) {
+            Ok(c) => fctrl(c),
+            Err(e) => last_err = Err(e),
+        }
+        match self.fan_control(4.into()) {
+            Ok(c) => fctrl(c),
+            Err(e) => last_err = Err(e),
```

Comment:
**question:** Is it okay to swallow the east(?) fan's error if west(?) also errors?

Reply:
To be honest, my rationale here was mostly just that existing code in this module will return the last error in cases where both fan controllers return an error. In practice, if both fan controllers return an error, the errors error will have been recorded elsewhere (in the I2C driver's ring buffer), so it's not *totally* thrown away...and the higher level code that calls this function ends up discarding the more detailed error information anyway. The caller of the `thermal` IPC interface is getting an error type that doesn't include details about the underlying I2C error condition regardless of what this function does.